Fixed quests 3908, 3972 & 3973

sky1683589933 opened the pull request:
Add quest execution script 3908 to fix the issue of not being able to complete quest properly, add quest execution script 3972 to fix empty dialog, and add quest execution script 3973 to fix the issue of completing quest without work items and encountering empty dialog.

sky1683589933 committed:
Add three new quest execution scripts

Add quest execution script 3908 to fix the issue of not being able to complete quest properly, add quest execution script 3972 to fix empty dialog, and add quest execution script 3973 to fix the issue of completing quest without work items and encountering empty dialog.

neon-dev commented on game-server/data/handlers/quest/sanctum/_3908ToMastertheDragon.java:
```diff
@@ -0,0 +1,65 @@
+
+public class _3908ToMastertheDragon extends AbstractQuestHandler {
+
+	private final static int questId = 3908;
+
+	public _3908ToMastertheDragon() {
+		super(questId);
+	}
```
```suggestion
	public _3908ToMastertheDragon() {
		super(3908);
	}
```
The static `questId` field here shadows the protected `questId` field from the AbstractQuestHandler, which should always be avoided.

neon-dev commented on game-server/data/handlers/quest/sanctum/_3908ToMastertheDragon.java:
```diff
@@ -0,0 +1,65 @@
+			if (targetId == 700515) {
+				switch (dialogActionId) {
+					case USE_OBJECT: 
+						if (player.getInventory().getItemCountByItemId(182206056) >= 1) 
+							removeQuestItem(env, 182206056, 1);
```
```suggestion
						if (removeQuestItem(env, 182206056, 1)) 
```

neon-dev commented on game-server/data/handlers/quest/sanctum/_3972TheGoBetween.java:
```diff
@@ -0,0 +1,63 @@
+
+public class _3972TheGoBetween extends AbstractQuestHandler {
+
+	private final static int questId = 3972;
+
+	public _3972TheGoBetween() {
+		super(questId);
+	}
```
```suggestion
	public _3972TheGoBetween() {
		super(3972);
	}
```

neon-dev commented on game-server/data/handlers/quest/sanctum/_3973LovesMessenger.java:
```diff
@@ -0,0 +1,101 @@
+
+public class _3973LovesMessenger extends AbstractQuestHandler {
+
+	private final static int questId = 3973;
+
+	public _3973LovesMessenger() {
+		super(questId);
+	}
```
```suggestion
	public _3973LovesMessenger() {
		super(3973);
	}
```

sky1683589933 committed:
Update game-server/data/handlers/quest/sanctum/_3908ToMastertheDragon.java

Co-authored-by: Neon <neon@beyond-aion.com>

sky1683589933 committed:
Update game-server/data/handlers/quest/sanctum/_3908ToMastertheDragon.java

Co-authored-by: Neon <neon@beyond-aion.com>

sky1683589933 committed:
Update game-server/data/handlers/quest/sanctum/_3972TheGoBetween.java

Co-authored-by: Neon <neon@beyond-aion.com>

sky1683589933 committed:
Update game-server/data/handlers/quest/sanctum/_3973LovesMessenger.java

Co-authored-by: Neon <neon@beyond-aion.com>

neon-dev committed:
Formatting and simpler handling

neon-dev commented:
Please take a look at these changes and tell me in case I missed something: 0290ea6b
I changed 3972 and 3973 to `<report_to_many>` handlers and fixed a problem in ReportToMany.java where the quest dialog after accepting the quest would not show.
The only difference between your 3973 Java handler and ReportToMany is that losing a workItem will not reset the quest step. This could be implemented in ReportToMany.java too, but we'd need info if and how the retail server handles this.

sky1683589933 commented:
> Please take a look at these changes and tell me in case I missed something: [0290ea6](https://github.com/beyond-aion/aion-server/commit/0290ea6bb66a70ecb8eb61bc99b69fcb80705166) I changed 3972 and 3973 to `<report_to_many>` handlers and fixed a problem in ReportToMany.java where the quest dialog after accepting the quest would not show. The only difference between your 3973 Java handler and ReportToMany is that losing a workItem will not reset the quest step. This could be implemented in ReportToMany.java too, but we'd need info if and how the retail server handles this.

Hi, I think Quest 3973 needs to have a rollback function without any work items, because the final result is to submit three work items. If this function is missing, all three work items can be thrown away to complete the quest. Of course, if the Quest step is executed normally, the rollback code will not be executed at all.

neon-dev commented:
Do you know what happens on official servers when you throw away a work item? Is the quest reset immediately, or when talking to the next npc, or only at the reward npc?

sky1683589933 commented:
> Do you know what happens on official servers when you throw away a work item? Is the quest reset immediately, or when talking to the next npc, or only at the reward npc?

No, It is impossible to remember or search for details that are ancient, and can only be inferred through reverse reasoning based on the results. Just my reasoning!
Unless I go to retail again, which is obviously unrealistic, haha! The details cannot be verified. The rollback is only for the purpose of completing the request without submitting the work item in exceptional circumstances. As for the logical step of rollback, I think it is best to immediately rollback when the work item is lost.

neon-dev commented:
> As for the logical step of rollback, I think it is best to immediately rollback when the work item is lost.

I agree.
Can I merge this pull request in its current state? The detection of discarded work items should be implemented separately, as it is not a small change and will affect all `<report_to_many>` quests.

sky1683589933 commented:
> I agree. Can I merge this pull request in its current state?
yeah， of course!

sky1683589933 commented:
> > I agree. Can I merge this pull request in its current state?
> > yeah， of course!

oh， I forgot to add an author for the new Java script execution. Although I don't care about this, it looks more formal this way, haha!

neon-dev commented:
It's your choice :octocat:

sky1683589933 commented:
Please help me add a name: sky123， thank you!

neon-dev commented on game-server/data/handlers/quest/sanctum/_3908ToMastertheDragon.java:
```diff
@@ -0,0 +1,62 @@
+import com.aionemu.gameserver.questEngine.model.QuestEnv;
+import com.aionemu.gameserver.questEngine.model.QuestState;
+import com.aionemu.gameserver.questEngine.model.QuestStatus;
+
```
```suggestion

/**
 * @author sky123
 */
```

sky1683589933 replied:
I hope your name is also included.
Because this is not something I did alone!

neon-dev replied:
I have stopped adding my name a while ago. Occasionally I add it to new files, but not every file.

neon-dev replied:
I have stopped adding my name a while ago. Occasionally I add it to new files I created, but not every file.

sky1683589933 replied:
ok， I respect your choice!

sky1683589933 committed:
Update game-server/data/handlers/quest/sanctum/_3908ToMastertheDragon.java

Co-authored-by: Neon <neon@beyond-aion.com>
